internal/isoutil/isoutil.go

```diff
@@ -228,3 +236,69 @@ func (h *installerHandler) VolumeIdentifier() (string, error) {
 
 	return strings.TrimSpace(string(volumeId)), nil
 }
+
+func GetFileLocation(filePath, isoPath string) (int64, error) {
+	d, err := diskfs.OpenWithMode(isoPath, diskfs.ReadOnly)
+	if err != nil {
+		return 0, err
+	}
+
+	fs, err := d.GetFilesystem(0)
+	if err != nil {
+		return 0, err
+	}
+
+	isoFile, err := fs.OpenFile(filePath, os.O_RDONLY)
+	if err != nil {
+		return 0, errors.Wrapf(err, "Failed to open file %s", filePath)
+	}
+
+	myFile := isoFile.(*iso9660.File)
+	defaultSectorSize := int64(2 * 1024)
+	offset := int64(myFile.Location()) * defaultSectorSize
+	return offset, nil
+}
+
+func WriteHeader(ignitionOffsetInfo, ramDiskOffsetInfo *OffsetInfo, isoPath string) error {
+	iso, err := os.OpenFile(isoPath, os.O_WRONLY, 0o664)
+	if err != nil {
+		return err
+	}
+	defer iso.Close()
+
+	headerOffset := int64(32768) // first 32KB in ISO
```

Comment:
This function is doing a lot of non-trivial stuff, some documentation would go a long way in helping to understand what's going on here. For example in this line, `// End of ISO 9660 system area`. And also `// The space at the end of the system area is likely to be empty, we use it to store the offsets of the ignition and ram disk files`. Assume the code readers don't have access to the jira ticket.

Reply:
Sure, added some comments the make it more explicit.